lib/fog/aliyun/requests/compute/create_security_group_egress_sg_rule.rb

```diff
@@ -0,0 +1,109 @@
+module Fog
+  module Compute
+    class Aliyun
+      class Real
+        # {Aliyun API Reference}[https://docs.aliyun.com/?spm=5176.100054.3.1.DGkmH7#/pub/ecs/open-api/securitygroup&authorizesecuritygroup]
+        def create_security_group_egress_sg_rule(securitygroup_id, dest_group_id, option={})
+          action   = 'AuthorizeSecurityGroupEgress'
+          sigNonce = randonStr()
+          time     = Time.new.utc
+
+          parameters = defalutParameters(action, sigNonce, time)
+          pathUrl    = defaultAliyunUri(action, sigNonce, time)
+          
+          parameters["SecurityGroupId"] = securitygroup_id
+          pathUrl += '&SecurityGroupId='
+          pathUrl += securitygroup_id
+
+          parameters["DestGroupId"] = dest_group_id
+          pathUrl += '&DestGroupId='
+          pathUrl += dest_group_id
+
+          nicType = 'intranet'
+          parameters["NicType"] = nicType
+          pathUrl += '&NicType='
+          pathUrl += nicType
+          
+          portRange = option[:portRange]
+          unless portRange
+            portRange = '-1/-1'
+          end
+          parameters["PortRange"] = portRange
+          pathUrl += '&PortRange='
+          pathUrl += URI.encode(portRange,'/[^!*\'()\;?:@#&%=+$,{}[]<>`" ')
+          
+          protocol = option[:protocol]
+          unless protocol
+            protocol = 'all'
+          end
+          parameters["IpProtocol"] = protocol
+          pathUrl += '&IpProtocol='
+          pathUrl += protocol
+
+          destGOAccount = option[:destGroupOwnerAccount]
+          if sourceGOAccount
+           parameters["DestGroupOwnerAccount"]=destGOAccount
+           pathUrl += '&DestGroupOwnerAccount='
+           pathUrl += destGOAccount
+          end
+
+          policy = option[:policy]
+          unless policy
+            policy = 'accept'
+          end
+          parameters["Policy"] = policy
+          pathUrl += '&Policy='
+          pathUrl += policy
+
+          priority = option[:priority]
+          unless priority
+            priority = '1'
+          end
+          parameters["Priority"] = priority
+          pathUrl += '&Priority='
+          pathUrl += priority
+
+          signature = sign(@aliyun_accesskey_secret, parameters)
+          pathUrl += '&Signature='
+          pathUrl += signature
```

Comment:
This url generation over here sounds like you need to work towards a method to do this based on the models.
I can see this is repeating for every route and is not a good thing. Maybe we could work towards the `as` option in the `attribute` method. Then get the result hash from `all_association_and_attributes` and turn it into this url.
What do you think?